app/scripts/metamask-controller.js

```diff
@@ -1526,6 +1542,22 @@ module.exports = class MetamaskController extends EventEmitter {
     }
   }
 
+  _onUnlock (keyRingControllerMemStore, cb) {
```

Comment:
I'm unclear what this addition does.

Reply:
Method receives a callback that is called whenever MetaMask is unlocked. Or, more precisely, whenever the `isUnlocked` property in the keyring controller store goes from false to true.

Passing this to a plugin allows it to do something when MetaMask is unlocked.